### PR TITLE
Added homework#17

### PR DESCRIPTION
**Pull request by YevheniiKubrak**

### CODE REVIEW

File: src/pageObjects/components/SignUpPopup.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        await expect(this._page).toHaveURL('https://qauto.forstudy.space/panel/garage')
+    }
+
+    async fillSignupForm({name, lastName, email, password, reenterPassword, switchFocus}){
+        await this.nameInput.fill(name)
+        await this.lastNameInput.fill(lastName)
+        await this.emailInput.fill(email)
+        await this.passwordInput.fill(password)
+        await this.reenterInput.fill(reenterPassword)
+        if (switchFocus) await this.nameInput.focus()
```
Comment (sttaran) [inline src/pageObjects/components/SignUpPopup.js:33]:
```js
if (switchFocus) await this.reenterInput.blur()
```

File: src/pageObjects/welcomePage/WelcomePage.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import SignUpPopup from "../components/SignUpPopup.js";
+
+
+export default class WelcomePage extends BasePage {
+    constructor(page) {
+        super(page, '/', page.locator('button', {hasText: 'Guest log in'}));
+    }
+
+    async openSignupPopup(){
+        const signUpButton = this._page.locator('button:text("Sign up")');
```
Comment (sttaran) [inline src/pageObjects/welcomePage/WelcomePage.js:12]: move it to constructor

File: tests/Auth/Signup/Signup.spec.js [hunk 3]
```diff
@@ -1,222 +1,143 @@
...
+        page = await context.newPage()
+        welcomePage = new WelcomePage(page)
+    })
+
+    test.beforeEach(async ()=>{
+        await welcomePage.open()
+        await welcomePage.waitLoaded()
+    })
+
     test('Successful Signup', async({page})=> {
```
Comment (sttaran) [inline tests/Auth/Signup/Signup.spec.js:27]:
```js
const password= faker.internet.password({length: 9, prefix: 'Qw1'})
const user = {
    name: '',
    lastName: faker.person.lastName(),
    email: faker.internet.email({firstName : 'aqa'}),
    password,
    reenterPassword: password
}
```

File: tests/Auth/Signup/Signup.spec.js [hunk 4]
```diff
@@ -1,222 +1,143 @@
...
     test('Should show error message when password do not match', async({page})=>{
         const user = {
-            firstName: faker.person.firstName(), 
+            name: faker.person.firstName(), 
             lastName: faker.person.lastName(),
             email: faker.internet.email({firstName : 'aqa'}),
             password: faker.internet.password({length: 9, prefix: 'Qw1'}),
-            reenterPassword: 'Qwertyuiopasdf1'
+            reenterPassword: 'Qwertyuiopasd1',
+            switchFocus: true
```
Comment (sttaran) [inline tests/Auth/Signup/Signup.spec.js:134]: switchFocus: true is not a part of user

Comment (sttaran) [reply tests/Auth/Signup/Signup.spec.js:134]: so better rename object

Comment (YevheniiKubrak) [reply tests/Auth/Signup/Signup.spec.js:134]: Should I rename every instance of "user" in every other test, or only in this instance?

**Review by sttaran — APPROVED** [review]

**Pull request merged**